featuregate/controller/Dockerfile

```diff
@@ -11,31 +11,32 @@ FROM $BUILDER_BASE_IMAGE as builder
 WORKDIR /workspace
 
 # Copy the Go Modules manifests
-COPY go.mod go.mod
-COPY go.sum go.sum
+COPY featuregate/controller/go.mod featuregate/controller/go.mod
```

Comment:
can we simplify this Dockerfile? this is what I've done for capabilities component https://github.com/vmware-tanzu/tanzu-framework/blob/14c8941150610f63985b149bb06a1b9742545bef/capabilities/controller/Dockerfile

Reply:
This is done specifically to preserve image layer caching in Docker and to avoid re-downloading Go modules every time. Comments in L22-L23 explains this.